Source/WebCore/editing/FrameSelection.cpp

```diff
@@ -2229,16 +2250,15 @@ void FrameSelection::updateAppearance()
     // If the caret moved, stop the blink timer so we can restart with a
     // black caret in the new location.
     if (caretRectChangedOrCleared || !shouldBlink || shouldStopBlinkingDueToTypingCommand(m_document.get()))
-        m_caretBlinkTimer.stop();
+        m_currentCaretAnimation->stop();
 
     // Start blinking with a black caret. Be sure not to restart if we're
     // already blinking in the right location.
-    if (shouldBlink && !m_caretBlinkTimer.isActive()) {
-        if (Seconds blinkInterval = RenderTheme::singleton().caretBlinkInterval())
-            m_caretBlinkTimer.startRepeating(blinkInterval);
+    if (shouldBlink && !m_currentCaretAnimation->isActive()) {
+        m_currentCaretAnimation->start(m_document->domWindow()->nowTimestamp());
 
-        if (!m_caretPaint) {
-            m_caretPaint = true;
+        if (!m_currentCaretAnimation->visible()) {
```

Comment:
Simpler to just call `m_currentCaretAnimation->setVisible(true)` unconditionally

Reply:
@smfr does this mean `invalidateCaretRect` should also be called unconditionally? Right now, this if statement is used so that the caret is only invalidated if its not already in the desired state

Reply:
I mean check the visibility state inside the call to setVisible() and early return.